### ISSUE TITLE
Do not suspend network reads in an HTTP/2 connection.

### ISSUE THREAD

**Issue by trustin** [line/armeria#1487]
Armeria currently suspends network reads when more than 128KiB are unconsumed by the stream subscribers. (See: [InboundTrafficController](https://github.com/line/armeria/blob/armeria-0.76.2/core/src/main/java/com/linecorp/armeria/internal/InboundTrafficController.java))

However, we should not disable network reads at all in an HTTP/2 connection because HTTP/2 has its own traffic control mechanism (`WINDOW_UPDATE` frame. See: [The Flow-Control Window](https://http2.github.io/http2-spec/#rfc.section.6.9.1)) Actually, arbitrarily disabling network reads can cause an unexpected permanent stall (i.e. dead lock) of an HTTP/2 connection because it can prevent the HTTP/2 implementation from reading a `WINDOW_UPDATE` frame.

Therefore, we should use `InboundTrafficController` only for HTTP/1 connections. We may also have some opportunity to gain additional performance thanks to this because 1) we do not need to suspend network reads and 2) we do not need to count the number of pending inbound bytes.

**Comment by trustin** [comment]
Finished implementing this but seeing some test failures related with streaming.

**Comment by trustin** [comment]
I realized that this is not going to work because Armeria will keep reading the traffic even if a subscriber consumes the traffic slowly, i.e. data will be buffered in the `DecodedHttp(Request|Response)` and eventually will overflow without `InboundTrafficController`.

However, while implementing this, I realized we are using smaller watermarks (128KiB) than initial connection-level window size (1MiB). We could use higher watermark values so that network reads are not suspended too often.
